src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -136,6 +137,24 @@ interface SwapProviderProps {
   children: ReactNode;
 }
 
+const determineIfUserQuickBought = (
+  quickBuyAnalyticalData: QuickBuyAnalyticalData | undefined,
+  inputAsset: ExtendedAnimatedAssetWithColors | ParsedSearchAsset | null,
+  outputAsset: ExtendedAnimatedAssetWithColors | ParsedSearchAsset | null
+) => {
+  if (!quickBuyAnalyticalData || !inputAsset || !outputAsset) {
+    return undefined;
+  }
+
+  const { assetUniqueId } = quickBuyAnalyticalData;
+
+  if (assetUniqueId !== inputAsset.uniqueId && assetUniqueId !== outputAsset.uniqueId) {
+    return undefined;
+  }
```

Comment:
I don't know how you'd get into this situation, but do we want to track this if the `assetUniqueId === inputAsset.uniqueId`? That would be a sale, right?

Maybe I am not fully following the logic, but it seems like you'd just wanted to ensure the output asset matches and bail otherwise.

Reply:
probably true, fixed